tests/test_utils.py

```diff
@@ -216,25 +216,56 @@ def test_failure(self, demo_func: Any, expected_traceback_regex: str) -> None:
         stderr_output = mock_stderr.read()
         assert_regex_match(expected_traceback_regex, stderr_output)
 
+    def test_simplified_traceback_with_no_module(self) -> None:
+        """
+        Test that simplified traceback logic can succeed even if
+        `inspect.getmodule(frame)` returns `None` for one of
+        the frames in the stacktrace.
+        """
+        demo_func = self.DemoFunctions.run_job_wrapper
+        expected_traceback_regex = dedent(
+            r"""
+            Traceback \(most recent call last\):$
+              File "[^"]+", line \d+, in nested_error$
+                assert False, "nested_err"$
+            AssertionError: nested_err$
+            assert False$
+            Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace\.$
+            """
+        )
+        mock_stderr = io.StringIO()
+        with raises(SystemExit, match="1"), patch("sys.stderr", new=mock_stderr):
+            # patch `inspect.getmodule` so that an exception will occur in the
+            # simplified traceback logic:
+            with patch("inspect.getmodule", new=lambda *args: None):
+                run_and_report(demo_func)
+        mock_stderr.seek(0)
+        stderr_output = mock_stderr.read()
+        assert_regex_match(expected_traceback_regex, stderr_output)
+
     def test_simplified_traceback_failure(self) -> None:
         """
         Test that a warning is printed and the original exception is re-raised
         when an exception occurs during the simplified traceback logic.
         """
         demo_func = self.DemoFunctions.run_job_wrapper
+
+        def throws(*args: Any, **kwargs: Any) -> None:
```

Comment:
`NoReturn` is more suitable than `None`, per https://docs.python.org/3/library/typing.html#typing.NoReturn and https://mypy.readthedocs.io/en/stable/more_types.html#the-noreturn-type.
Not important, just a minor suggestion.